src/components/search-package-list-item/search-package-list-item.test.js

```diff
@@ -0,0 +1,118 @@
+import React from 'react';
+import { screen, render, fireEvent } from '@testing-library/react';
+
+import { Router } from 'react-router';
+import { createMemoryHistory } from 'history';
+
+import SearchPackageListItem from './search-package-list-item';
+
+jest.mock('../tags-label', () => () => (<div>TagsLabel component</div>));
+jest.mock('../hidden-label', () => () => (<div>HiddenLabel component</div>));
+
+const routerHistory = createMemoryHistory();
+
+const testTags = ['first', 'second', 'third'];
+
+const testEmptyProps = {};
+
+const testBasicProps = {
+  item: {
+    name: 'test-package-name',
+    providerName: 'test-provider-name',
+    isSelected: true,
+    titleCount: 1,
+    visibilityData: {
+      isHidden: false,
+    },
+    tags: {
+      tagList: testTags,
+    },
+  },
+  link: 'test-link-url',
+};
+
+const testFullProps = {
+  ...testBasicProps,
+  item: {
+    ...testBasicProps.item,
+    visibilityData: {
+      isHidden: true,
+    }
+  },
+  showProviderName: true,
+  showTitleCount: true,
+  showTags: true,
+};
+
+describe('Given SearchPackageListItem', () => {
+  const handleClick = jest.fn();
+
+  const renderSearchPackageListItem = ({ ...props }) => render(
+    <Router history={routerHistory}>
+      <SearchPackageListItem {...props} />
+    </Router>
+  );
+
+  it('should render skeleton when item is not provided', () => {
+    renderSearchPackageListItem(testEmptyProps);
+
+    const isSkeleton = document
```

Comment:
Instead of querySelector you should use returned selectors from `renderSearchPackageListItem` and select items by their text or `data-testid` attribute
Here's docs about using RTL queries https://testing-library.com/docs/queries/about/

Reply:
Please, take a look how tests wrote for other components and try to write them in the same template.